src/components/algorithm/EditAlgorithm.js

```diff
@@ -0,0 +1,248 @@
+import SaveIcon from '@material-ui/icons/Save';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+
+import Editor from '../common/Editor';
+import Loader from '../common/Loader';
+import Main from '../common/Main';
+import { addAlgorithm, getAlgorithm, saveAlgorithm } from '../../actions';
+import { AUTHOR_GRAASP, AUTHOR_USER } from '../../config/constants';
+
+const styles = (theme) => ({
+  button: {
+    margin: theme.spacing(1),
+  },
+  infoAlert: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+});
+
+class EditAlgorithm extends Component {
+  state = {
+    name: '',
+    description: '',
+    code: '',
+  };
+
+  static propTypes = {
+    algorithm: PropTypes.instanceOf(Map).isRequired,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    history: PropTypes.shape({
+      goBack: PropTypes.func.isRequired,
+    }).isRequired,
+    dispatchGetAlgorithm: PropTypes.func.isRequired,
+    dispatchSaveAlgorithm: PropTypes.func.isRequired,
+    dispatchAddAlgorithm: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      button: PropTypes.string.isRequired,
+      infoAlert: PropTypes.string.isRequired,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+    activity: PropTypes.bool.isRequired,
+  };
+
+  componentDidMount() {
+    const {
+      dispatchGetAlgorithm,
+      match: {
+        params: { id },
+      },
+    } = this.props;
+
+    dispatchGetAlgorithm({ id });
+  }
+
+  componentDidUpdate({ algorithm: prevAlgorithm }) {
+    const { algorithm } = this.props;
+    if (algorithm && algorithm !== prevAlgorithm) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        name: algorithm.get('name'),
+        description: algorithm.get('description'),
+        code: algorithm.get('code'),
+      });
+    }
+  }
+
+  handleNameOnChange = (event) => {
+    this.setState({ name: event.target.value });
+  };
+
+  handleDescriptionOnChange = (event) => {
+    this.setState({ description: event.target.value });
+  };
+
+  handleCodeOnChange = (code) => {
+    this.setState({ code });
+  };
+
+  handleBack = () => {
+    const {
+      history: { goBack },
+    } = this.props;
+    goBack();
+  };
+
+  handleSave = () => {
+    const {
+      dispatchSaveAlgorithm,
+      dispatchAddAlgorithm,
+      algorithm,
+    } = this.props;
+    const { name, description, code } = this.state;
+    if (algorithm && name) {
+      const author = algorithm.get('author');
+
+      if (author === AUTHOR_GRAASP) {
+        // add as a new algorithm instead
+        const payload = { name, description, author: AUTHOR_USER, code };
+        dispatchAddAlgorithm(payload);
+      } else {
+        const metadata = { ...algorithm?.toObject(), name, description };
+        dispatchSaveAlgorithm({ metadata, code });
+      }
+    }
+  };
+
+  renderBackButton = () => {
```

Comment:
Maybe we should have a back button component if possible aha. What do you think?

Reply:
Ah good call!